burr/core/state.py

```diff
@@ -196,22 +235,31 @@ def get_all(self) -> Dict[str, Any]:
     def serialize(self, **kwargs) -> dict:
         """Converts the state to a JSON serializable object"""
         _dict = self.get_all()
-        return {
-            # TODO: handle field specific custom serialization
-            k: serde.serialize(v, **kwargs)
-            for k, v in _dict.items()
-        }
+
+        def _choose_serde(k, v, **extrakwargs) -> Union[dict, str]:
+            """chooses the correct serde function for the given key"""
+            if k in FIELD_SERIALIZATION:
+                result = FIELD_SERIALIZATION[k][0](v, **extrakwargs)
+                if not isinstance(result, dict):
+                    raise ValueError(
+                        f"Field serde for {k} must return a dict, but return {type(result)}"
```

Comment:
print the actual result too

Reply:
why? it's either None, or not a dictionary...

Reply:
It helps debugging -- I've had this exact same type of error before, and my first step is always to figure what the actual result is. Not strictly necessary, but gives the context users need.

Reply:
sure, fine. limiting size of str.